libfwupdplugin/fu-fmap-firmware.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2020 Benson Leung <bleung@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include "fu-firmware.h"
+
+#define FMAP_SIGNATURE		"__FMAP__"
+#define FMAP_VER_MAJOR		1	/* this header's FMAP major version */
+#define FMAP_VER_MINOR		1	/* this header's FMAP minor version */
+#define FMAP_STRLEN		32	/* maximum length for strings, */
+					/* including null-terminator */
+#define FMAP_AREANAME		"FMAP"
+
+#define FU_TYPE_FMAP_FIRMWARE (fu_fmap_firmware_get_type ())
+G_DECLARE_FINAL_TYPE (FuFmapFirmware, fu_fmap_firmware, FU, FMAP_FIRMWARE, FuFirmware)
```

Comment:
FINAL is _okay_, but this means you can never create a subclass of FuFmapFirmware. If you think they'll ever be different types of FuFmapFirmware e.g. FuLegacyFmapFirmware or FuSuperFmapFirmware this should probably be DERIVED. If the latter is true, I can help you convert the various `self`'s to `priv`s if you want.

Reply:
Well specifically CrosEC is a type of fmap firmware per my understanding.  So it shouldn't be final then.

Reply:
Sounds good. Will fix.

Reply:
The good news is that the fu-fmap-firmware was so simple at this point that it didn't use any `self`s. New version will use `G_DECLARE_DERIVABLE_TYPE ` instead.

Reply:
Done.